src/main/scala/org/ergoplatform/nodeView/history/storage/modifierprocessors/ToDownloadProcessor.scala

```diff
@@ -94,12 +94,13 @@ trait ToDownloadProcessor extends ScorexLogging {
   }
 
   private def requiredModifiersForHeader(h: Header): Seq[(ModifierTypeId, ModifierId)] = {
+    lazy val emptyExtensionId: ModifierId = Extension(h.id).id
     if (!config.verifyTransactions) {
       Seq.empty
     } else if (config.stateType.requireProofs) {
-      Seq((BlockTransactions.modifierTypeId, h.transactionsId), (ADProofs.modifierTypeId, h.ADProofsId))
+      h.sectionIds.filter(id => !(id._2 sameElements emptyExtensionId))
```

Comment:
`filterNot` could be more straightforward here

Reply:
fixed